### PR TITLE
README documentation Part 2

### PR DESCRIPTION
**Pull request by pablisco**
WIP plus depends on #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,338 +87,701 @@ We provide you with a sample token provider implementation. You can enable / dis
...
+If you are using coroutines this can be wrapped into a suspending method like this:
+
+```kotlin
+suspend fun ChatManager.connectForUser(): Result<CurrentUser, Error> = 
+  suspendCoroutine{ c -> c.resume(connect().wait()) }
+  
+// or, if want to treat error as an exception:
+
+suspend fun ChatManager.connectForUser(): CurrentUser = suspendCoroutine { c ->
+  connect().wait().let { result ->
```
[the remark this answers is not in the document]
Comment (pablisco) [reply README.md]: Trying to user 2 spaces, as it's easier to render on github

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
+import com.pusher.platform.network.toFuture
 import com.pusher.platform.tokenProvider.TokenProvider
-import com.pusher.util.Result
-import com.pusher.util.mapResult
+import com.pusher.util.*
 import elements.Error
+import elements.Subscription
 import java.util.concurrent.Future
 
 class CurrentUser(
```
[the remark this answers is not in the document]
Comment (pablisco) [reply chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt]: The JS client doesn't have these values:
https://docs.pusher.com/chatkit/reference/javascript#user-properties

File: README.md [hunk 3]
```diff
@@ -46,338 +87,701 @@ We provide you with a sample token provider implementation. You can enable / dis
...
+### Access other user's cursors
+
+After subscribing to a room, read cursors for members of that room can be accessed by supplying a `userId` as the second parameter to the `readCursor` method.
+
+```kotlin
+currentUser.readCursor(
+  roomId: romroomId,
+  userId: "alice"
+).wait().let { result -> // Future<Result<Boolean, Error>>
+  when(result) { // Result<Int, Error>, either the new message id or an error
```
[the remark this answers is not in the document]
Comment (pablisco) [reply README.md]: it was a copy paste :D

File: docs/Result.md [hunk 4]
```diff
@@ -1,4 +1,31 @@
-# TODO
+# Result<A, B>
+
+This data type follow the `Either` functional pattern to return one of two possibilities with some renames to add meaning and extensions to work with futures. All the endpoints that deliver results in the SDK user some type `A` and an `Error` (different from `kotlin.Error`) as associated types.
```
Comment (hamchapman) [inline docs/Result.md]: > follow

->

> follows

**Pull request merged**